install_config/certificate_customization.adoc

```diff
@@ -29,72 +29,300 @@ custom certificates can be configured using the
+
+----
+openshift_hosted_router_certificate={"certfile": "/path/to/apps.c1-ocp.myorg.com.crt", "keyfile": "/path/to/apps.c1-ocp.myorg.com.key", "cafile": "/path/to/apps.c1-ocp.myorg.com.ca.crt"}
+----
+
+For example:
+
+----
+openshift_hosted_router_certificate={"certfile": "/home/cloud-user/star-apps.148.251.233.173.nip.io.cert.pem", "keyfile": "/home/cloud-user/star-apps.148.251.233.173.nip.io.key.pem", "cafile": "/home/cloud-user/ca-chain.cert.pem"}
+----
+
+Where the parameter values are: 
+
+* *certfile* contains the {product-title} router certificate.
+
+* *keyfile* contains the {product-title} wildcard key
+
+* *cafile* contains the root CA for this key and certificate. If an intermediate CA is in use it should contain both the intermediate and root CA.
+
+If these certificate files are new to your {product-title} cluster, run an Ansible playbook to add these files to the {product-title} configuration files. 
+The playbook adds the certificate files to the *_/etc/origin/master/named_certificates/_* directory.
+
+----
+ifdef::openshift-enterprise[]
+# ansible-playbook [-i /path/to/inventory] \
+    /usr/share/ansible/openshift-ansible/playbooks/deploy_cluster.yml
+endif::[]
+ifdef::openshift-origin[]
+# ansible-playbook [-i /path/to/inventory] \
+    ~/openshift-ansible/playbooks/deploy_cluster.yml
+endif::[]
+----
+
+If xref:../install_config/redeploying_certificates.adoc#redeploying-all-certificates-current-ca[the certificates are not new], 
+for example, you want to change existing certificates or replace expired certificates, run the following playbook:
+
+----
+ansible-playbook /usr/share/ansible/openshift-ansible/playbooks/redeploy-certificates.yml
+----
+
+[NOTE] 
+For this playbook to run, the certificate names must not change. If the certificate names change, rerun the Ansible `config.yml` playbook
+as if the certificates were new.
+
+[[configuring-custom-certificates-registry]]
+== Configuring a Custom Certificate for the Image Registry
+
+The {product-title} image registry is an internal service that facilitates builds and deployments. Most of the communication with the registry is handled by internal components in {product-title}. As such, there should be no need to replace the certificate used by the registry service itself. 
+
+However, by default, the registry uses routes to allow external systems and users to do pulls and pushes of images. You can use a _re-encrypt route_ with a custom certificate that will be presented to external users instead of using the internal, self-signed certificate. 
+
+To configure this, add the xref:../install_config/install/advanced_install.adoc#advanced-install-configuring-docker-route[following lines] 
+to the Ansible inventory file, by default *_/etc/ansible/hosts_* file, in the `[OSEv3:vars]`
```

Comment:
"to the `[OSEv3:vars]` section of the Ansible inventory file, by default *_/etc/ansible/hosts_* file. Specify the certificates to use with the registry route."